CONTRIBUTING.md

```diff
@@ -1,45 +1,111 @@
+
+## Testing
+
+Much of the work of maintaining jrnl involves testing rather than coding.
+
+The nature of jrnl means we deal with extremely sensitive data, and can't risk data loss. While jrnl does have a comprehensive automated testing suite, user testing is crucial to mitigating this risk.
+
+### Prereleases
+
+[Prereleases are deployed through PyPi much like normal releases](https://pypi.org/project/jrnl/#history). You can use [pipx](https://pypi.org/project/pipx/) to fetch them and test them. See the [changelog](CHANGELOG.md) for information on what has changed with each release.
+
+### Pull requests
+
+If you are comfortable enough with git, you're encouraged to fetch particular [pull requests](https://github.com/jrnl-org/jrnl/pulls), test them yourself, and report back your findings. Bonus points if you can add a screencast of how the new feature works.
+
+### Confirm bug reports
+
+There are always [open bugs among our GitHub issues](https://github.com/jrnl-org/jrnl/issues?q=is%3Aissue+is%3Aopen+label%3Abug) and many are specific to a particular OS, Python version, or jrnl version. A simple comment like "Confirmed on jrnl v2.2, MacOS 10.15, Python 3.8.1" would be extremely helpful in tracking down bugs.
+
+### Automate tests
+
+See the develop section below for information on how to contribute new automated tests.
+
+## Submitting feature requests and ideas
+
+If you have a feature request or idea for jrnl, please [open a new issue](https://github.com/jrnl-org/jrnl/issues/new/choose) and describe the goal of the feature, and any relevant use cases. We'll discuss the issue with you, and decide if it's a good fit for the project.
 
 When discussing new features, please keep in mind our design goals. jrnl strives to do one thing well. To us, that means:
 
 * be _slim_
 * have a simple interface
 * avoid duplicating functionality
 
-## New programmers and programmers new to python
+## Developing
 
-Although jrnl has grown quite a bit since its inception, the overall complexity (for an end-user program) is fairly low, and we hope you'll find the code easy enough to understand.
+### Getting your environment set up
 
-If you have a question, please don't hesitate to ask! Python is known for its welcoming community and openness to novice programmers, so feel free to fork the code and play around with it! If you create something you want to share with us, please create a pull request. We never expect pull requests to be perfect, idiomatic, instantly mergeable code. We can work through it together!
+You will need to install [poetry](https://poetry.eustace.io/) to develop jrnl. It will take care of all of the project's other dependencies, except for [mkdocs](https://www.mkdocs.org), which you must install if you wish to preview documentation changes.
```

Comment:
I'm pretty sure mkdocs isn't a separate requirement. But maybe I'm wrong? Did you test this?

Reply:
You're right! I just ensured I didn't have mkdocs installed globally and successfully served the docs. Will edit.